src/main/java/frc/trigon/robot/subsystems/collector/Collector.java

```diff
@@ -0,0 +1,46 @@
+package frc.trigon.robot.subsystems.collector;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Collector extends SubsystemBase {
+
+    private static final WPI_TalonFX
+            openingMotor = CollectorConstants.OPENER_MOTOR,
+            collectingMotor = CollectorConstants.COLLECTOR_MOTOR;
+
+    private final static Collector INSTANCE = new Collector();
+
+    private Collector() {
+
+    }
+
+    public static Collector getInstance() {
+        return INSTANCE;
+    }
+
+    private void stop() {
+        collectingMotor.disable();
+        openingMotor.disable();
+    }
+
+    private void collect() {
+        openingMotor.set(CollectorConstants.OPENING_VOLTAGE);
+        collectingMotor.set(ControlMode.PercentOutput, CollectorConstants.COLLECTING_VOLTAGE);
+    }
+
+    private void close() {
+        openingMotor.set(CollectorConstants.CLOSING_VOLTAGE);
+        collectingMotor.disable();
+    }
+
+    private void eject() {
+        collectingMotor.set(ControlMode.PercentOutput, CollectorConstants.EJECTING_VOLTAGE);
+    }
+
+    public boolean isOpen() {
+        return openingMotor.get() > 0;
```

Comment:
`.get()` returns the velocity of the motor not the position. you can't know if its opened or closed from the velocity.

Reply:
It returns the last requested percent output. If positive, the system is trying to open which is good enough.

Reply:
It's not good enough. if it is open fully the requested output should be 0 however the collector is open.

Reply:
Nope, there's always power to the opener. When closed it's negative, and positive when open.

Reply:
you don't have to have power to keep it open... unless they changed the intake, which now that I think of it, I hope they did....

Reply:
Yes you do, since it's the simplest way to keep it open/closed

Reply:
It will stall the motor if you keep putting power and it is fully open or closed and stopped mechanically from moving more

Reply:
That's the reason we put the current limit.
That's how it was done in the season, and that's how it was done in 2020.

Reply:
fine